src/loaddb/load_object.c

```diff
@@ -486,11 +486,11 @@ text_print (TEXT_OUTPUT * tout, const char *buf, int buflen, char const *fmt, ..
       va_end (ap);
     }
 
-  if (nbytes > 0 && buf != NULL)
+  if (nbytes > 0)
     {
       if (nbytes < size)
 	{			/* OK */
-	  if (buflen > 0)
+	  if (buflen > 0 && buf != NULL)
 	    {			/* unformatted print */
```

Comment:
It seems to change the existing semantic. The changed code skips `memcpy()` if buf  is NULL, but it just assumed `buf` is not NULL if `buflen > 0` in the previous code. How about this below:
```suggestion
	  if (buflen > 0)
	    {			/* unformatted print */
	      assert (buf != NULL);
```

Reply:
I'm not sure if assert() is better than just null point exception, and it can pass cppcheck.

Reply:
I think passing a null pointer to `memcpy()` is unsafe according to the following reference.

https://en.cppreference.com/w/cpp/string/byte/memcpy
> If either dest or src is an invalid or null pointer, the behavior is undefined, even if count is zero.

Here is another discussion about passing the null pointer argument to `memcpy ()`.
https://stackoverflow.com/questions/5243012/is-it-guaranteed-to-be-safe-to-perform-memcpy0-0-0

Reply:
Yes. It has to be avoided, so assert it. If it is skipped because `buf` is null, it may behave unexpectedly.

Reply:
Sorry for not spotting it either.
This function has an input invariant that looks like the folowing:
```
assert((buf == NULL && buflen ==0) || (buf != NULL && buflen > 0));
```

Also, I would also add a new indirection for this function in the form (pseudocode):
```
int
text_print_fixed_buffer (TEXT_OUTPUT * tout, const char *buf, char const *fmt, ...)
{
  assert(buf != NULL);
  return (tout, buf, strlen(buf), fmt, ...);
}

```
because there are some issues.
For example there is one usage which I think is wrong (~1070):
```
	  if (i < len - 1)
	    {
	      CHECK_PRINT_ERROR (text_print (tout, ",\n ", 2, NULL));
	    }
```
and usage of the type:
```
	  CHECK_PRINT_ERROR (text_print
			     (tout, (left_nbytes <= 0) ? "'\'+\n \'" : "'", (left_nbytes <= 0) ? 6 : 1, NULL));
```
is error prone.
But maybe this is not for this PR.